pkg/process/checks/process_linux.go

```diff
@@ -20,49 +57,170 @@ func (p *ProcessCheck) useWLMCollection() bool {
 
 // processesByPID returns the processes by pid from different sources depending on the configuration (process probe or workloadmeta)
 // workload meta process collection is only available on linux and TODO: will eventually be the only source for linux process collection
-func (p *ProcessCheck) processesByPID(collectStats bool) (map[int32]*procutil.Process, error) {
+func (p *ProcessCheck) processesByPID() (map[int32]*procutil.Process, error) {
 	if p.useWLMProcessCollection {
 		wlmProcList := p.wmeta.ListProcesses()
 		pids := make([]int32, len(wlmProcList))
 		for i, wlmProc := range wlmProcList {
 			pids[i] = wlmProc.Pid
 		}
 
-		statsForProcess := make(map[int32]*procutil.Stats)
-		if collectStats {
-			var err error
-			statsForProcess, err = p.probe.StatsForPIDs(pids, p.clock.Now())
-			if err != nil {
-				return nil, err
-			}
+		statsForProcess, err := p.probe.StatsForPIDs(pids, p.clock.Now())
+		if err != nil {
+			return nil, err
 		}
 
 		// map to common process type used by other versions of the check
 		procs := make(map[int32]*procutil.Process, len(wlmProcList))
 		for _, wlmProc := range wlmProcList {
-			procs[wlmProc.Pid] = mapWLMProcToProc(wlmProc, statsForProcess[wlmProc.Pid])
+			stats, exists := statsForProcess[wlmProc.Pid]
+			// we need to check if the stats exist because there can be a lag between when a process is stored into WLM and when we query for its stats
+			// ex. a process is stopped but still exists in WLM, so the stats don't exist, and we shouldn't report it anymore
+			if !exists {
+				continue
+			}
+			procs[wlmProc.Pid] = mapWLMProcToProc(wlmProc, stats)
```

Comment:
Let's also check the create time from the stats against the create time from WLM, to ensure its actually the same process and not another with the same pid.